### PR TITLE
parallel xval with svm

### PR DESCRIPTION
**Pull request by Saurabh7**
NOT merge ready

### CODE REVIEW

**Comment by Saurabh7** [comment]
This is specific to LibSVM * , since I am creating its different objects in each fold.

For working with say `Machine*` , I was thinking on the lines of making a virtual method `create_shallow_copy` to create Machine objects with similar parameters.

Similar as above for Features. Currently using a `create_shallow_copy` for `DenseFeatures<float64_t>`

File: src/shogun/evaluation/CrossValidation.cpp [hunk 1]
```diff
@@ -264,8 +272,27 @@ float64_t CCrossValidation::evaluate_one_run()
 		m_machine->set_store_model_features(true);
 
 		/* do actual cross-validation */
+#pragma omp parallel for
```
Comment (karlnapf) [inline src/shogun/evaluation/CrossValidation.cpp]: I think you want to parallelise over both folds and runs

File: src/shogun/evaluation/CrossValidation.cpp [hunk 2]
```diff
@@ -264,8 +272,27 @@ float64_t CCrossValidation::evaluate_one_run()
...
 		for (index_t i=0; i <num_subsets; ++i)
 		{
+			CMachine* machine;
```
Comment (karlnapf) [inline src/shogun/evaluation/CrossValidation.cpp]: OK
I suggest for now: `CSGObject::clone`, which gives a deep copy.
Try that and see how bad it is ;)

Comment (karlnapf) [inline src/shogun/evaluation/CrossValidation.cpp]: We copy the feature matrix this way, but that is OK for problems that are not too big. Let's see whether this works.
We can then start optimising further

**Pull request closed without merging**